### PR TITLE
e2e ginkgo timeout fixes

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind bug
/kind failing-test

#### What this PR does / why we need it:

After https://github.com/kubernetes/kubernetes/pull/112923 was merged, some tests started to fail.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/assign @aojea

### CODE REVIEW

File: test/e2e/kubectl/kubectl.go [hunk 1]
```diff
@@ -579,7 +579,10 @@ var _ = SIGDescribe("Kubectl client", func() {
 				if !strings.Contains(ee.String(), "timed out") {
 					framework.Failf("Missing expected 'timed out' error, got: %#v", ee)
 				}
-				framework.ExpectNoError(e2epod.WaitForPodToDisappear(ctx, f.ClientSet, ns, "failure-3", labels.Everything(), 2*time.Second, wait.ForeverTestTimeout))
+				// A single wait.ForeverTestTimeout = 30s was
+				// too short and usually timed out. Found after
+				// df5d84ae811b added error checking.
+				framework.ExpectNoError(e2epod.WaitForPodToDisappear(ctx, f.ClientSet, ns, "failure-3", labels.Everything(), 2*time.Second, 2*wait.ForeverTestTimeout))
```
Comment (aojea) [inline test/e2e/kubectl/kubectl.go:585]:
```suggestion
				framework.ExpectNoError(e2epod.WaitForPodToDisappear(ctx, f.ClientSet, ns, "failure-3", labels.Everything(), 2*time.Second, 2*v1.DefaultTerminationGracePeriodSeconds*time.Second))
```

this depends on the gracedperiod IIUIC

Comment (aojea) [reply test/e2e/kubectl/kubectl.go:585]: I've verified that this works

**Comment by aojea** [comment]
just one change and we are good to go

File: test/e2e/kubectl/kubectl.go (hunk 1 above)
Comment (pohly) [reply test/e2e/kubectl/kubectl.go]: Changed. I also removed my comment because it no longer applied.

**Comment by aojea** [comment]
/lgtm
/approve

Thanks

**Pull request merged**